src/scheduler/check.cpp

```diff
@@ -1566,13 +1570,36 @@ check_normal_node_path(status *policy, server_info *sinfo, queue_info *qinfo, re
 			/* if there are nodes assigned to the queue, then check those */
 			if (qinfo->has_nodes)
 				ninfo_arr = qinfo->nodes;
-			else
-				ninfo_arr = sinfo->unassoc_nodes;
-		} else
-			/* last up we're not in a queue with nodes -- use the unassociated nodes */
-			ninfo_arr = sinfo->unassoc_nodes;
+			else if (sinfo->svr_to_psets != NULL) {	/* Multi-server psets */
+				int i;
+
+				/* Find the pset of the server which owns the job */
+				for (i = 0; sinfo->svr_to_psets[i] != NULL; i++) {
+					if (strcmp(sinfo->svr_to_psets[i]->svr_inst_id, resresv->job->svr_inst_id) == 0) {
+						msvr_pset[0] = sinfo->svr_to_psets[i]->np;
+						if (!resresv->job->is_subjob && spec->total_chunks == 1) {
```

Comment:
I think is_subjob won't be true here, because subjobs are created in run_update_resresv. So you might want to check for is_array parent instead.

Reply:
thanks for catching this and suggesting an alternative